src/main/java/com/mcmoddev/mmdbot/modules/logging/misc/ScamDetector.java

```diff
@@ -0,0 +1,130 @@
+            final var mutedRoleID = MMDBot.getConfig().getRole("muted");
+            final var embed = getLoggingEmbed(msg, "");
+            msg.delete().queue($ -> {
+                getLoggingChannel(guild).sendMessageEmbeds(embed).queue();
+                // TODO once JDA is updated, maybe timeout the user. It seems a better idea
+                guild.addRoleToMember(member, guild.getRoleById(mutedRoleID)).queue();
+            });
+        }
+    }
+
+    @Override
+    public void onGuildMessageUpdate(@NotNull final GuildMessageUpdateEvent event) {
+        final var msg = event.getMessage();
+        if (containsScam(msg)) {
+            final var member = msg.getMember();
+            final var msgChannel = msg.getTextChannel();
+            final var guild = msg.getGuild();
+            final var mutedRoleID = MMDBot.getConfig().getRole("muted");
+            final var embed = getLoggingEmbed(msg, ", by editing an old message");
+            msg.delete().queue($ -> {
+                getLoggingChannel(guild).sendMessageEmbeds(embed).queue();
+                // TODO once JDA is updated, maybe timeout the user. It seems a better idea
+                guild.addRoleToMember(member, guild.getRoleById(mutedRoleID)).queue();
+            });
+        }
+    }
+
+    private static MessageEmbed getLoggingEmbed(final Message message, final String extraDescription) {
+        final var member = message.getMember();
+        return new EmbedBuilder().setTitle("Scam link detected!")
+            .setDescription(String.format("User %s sent a scam link in %s%s! Their message was deleted, and they were muted!", member.getAsMention(),
+                message.getTextChannel().getAsMention(), extraDescription))
+            .addField("Message Content", "> " + message.getContentRaw(), false)
+            .setColor(Color.RED)
+            .setTimestamp(Instant.now())
+            .setFooter("User ID: " + member.getIdLong())
+            .setThumbnail(member.getEffectiveAvatarUrl()).build();
+    }
+
+    private static TextChannel getLoggingChannel(final Guild guild) {
+        return guild.getTextChannelById(MMDBot.getConfig().getChannel("events.requests_deletion"));
+    }
+
+    public static boolean containsScam(final Message message) {
+        final String msgContent = message.getContentRaw().toLowerCase();
+        for (final var link : SCAM_LINKS) {
+            if (msgContent.contains(link)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static void setupScamLinks() {
+        MMDBot.LOGGER.debug("Setting up scam links! Receiving data from {}.", SCAM_LINKS_DATA_URL);
+        try (var is = new URL(SCAM_LINKS_DATA_URL).openStream()) {
+            final String result = new String(is.readAllBytes(), StandardCharsets.UTF_8);
+            GSON.fromJson(result, JsonArray.class)
+                .forEach(link -> SCAM_LINKS.add(link.getAsString()));
```

Comment:
This will be comparatively slow as opposed to using `SCAM_LINKS.addAll`, since each `add` operation will cause a synchronization to happen.

Additionally, the list should probably be cleared before adding the new links. Perhaps it may be more beneficial to synchronize on our own mutex/`Semaphore`/`ReadWriteLock` instance rather than relying on `Collections.synchronizedList`.